pkg/customresourcestate/registry_factory.go

```diff
@@ -237,9 +287,41 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 				sValueFrom[0] == '[' && sValueFrom[len(sValueFrom)-1] == ']' &&
 				// "[...]" and not "[]".
 				len(sValueFrom) > 2 {
+				// remove the brackets
 				extractedValueFrom := sValueFrom[1 : len(sValueFrom)-1]
-				if key == extractedValueFrom {
-					gotFloat, err := toFloat64(it, c.NilIsZero)
+				// search space to resolve the dynamic valueFrom from the wildcard labels
+				dynamicValueFromScope := c.compiledCommon.labelFromPath
+				lastUnderscoreIndex := strings.LastIndex(extractedValueFrom, "_")
+				if lastUnderscoreIndex != -1 {
+					// For:
+					// labelsFromPath:
+					//  "foo_*": ["spec", "fooObj"]
+					unresolvedKey := extractedValueFrom[:lastUnderscoreIndex] + "_*"
+					dynamicPaths, ok := dynamicValueFromScope[unresolvedKey]
+					if ok {
+						var resolvedKeyArr []string
+						for _, dynamicPath := range dynamicPaths {
+							resolvedKeyArr = append(resolvedKeyArr, dynamicPath.part)
+						}
+						// resolvedKey will map to unresolved key "foo_*"'s corresponding valid path string, i.e., "spec_fooObj_*".
+						resolvedKey := strings.Join(resolvedKeyArr, "_")
+						extractedValueFrom = resolvedKey + extractedValueFrom[lastUnderscoreIndex:]
+					}
+				}
```

Comment:
Why do we do that?

As far as I can tell this is made to propagate the wildcard to the value no? Like:
```
"foo_*": ["spec", "fooObj"]
```
yields the following path:
```
spec_fooObj_*
```
no?